tests/test_activate.py

```diff
@@ -3099,3 +3099,110 @@ def test_activate_and_deactivate_for_uninitialized_env(conda_cli):
     assert conda_error.value.message.startswith(
         "Run 'conda init' before 'conda deactivate'"
     )
+
+
+# The MSYS2_PATH tests are slightly unusual in two regards: firstly
+# they stat(2) for potential directories which indicate which (of
+# several) possible MSYS2 environments have been installed; secondly,
+# conda will pass a Windows pathname prefix but conda-build will pass
+# a Unix pathname prefix (in particular, an MSYS2 pathname).
+
+
+def assert_MSYS2_PATH(create_dirs=None, exp_paths=None, no_paths=None):
+    with tempdir() as td:
+        mkdir_p(join(td, "conda-meta"))
+        activate_d_dir = mkdir_p(join(td, "etc", "conda", "activate.d"))
+        activate_d_1 = join(activate_d_dir, "see-me.sh")
+        activate_d_2 = join(activate_d_dir, "dont-see-me.bat")
+        touch(join(activate_d_1))
+        touch(join(activate_d_2))
+
+        activate_env_vars = join(td, PREFIX_STATE_FILE)
+        with open(activate_env_vars, "w") as f:
+            f.write(ENV_VARS_FILE)
+
+        write_pkg_env_vars(td)
+
+        library_dir = join(td, "Library")
+        mkdir_p(dirname(library_dir))
+
+        if create_dirs:
+            for dir in create_dirs:
+                mkdir_p(dirname(join(library_dir, dir, "bin")))
+
+        # td is a Windows pathname
+        ce_act = CmdExeActivator()
+        builder = ce_act.build_activate(td)
+        paths = ce_act._replace_prefix_in_path(td, td)
+
+        if exp_paths:
+            for dir in exp_paths:
+                exp_dir = join(library_dir, dir, "bin")
+                assert ce_act.path_conversion(exp_dir) in paths
+
+        if no_paths:
+            for dir in no_paths:
+                exp_dir = join(library_dir, dir, "bin")
+                assert ce_act.path_conversion(exp_dir) not in paths
+
+        ps_act = PowerShellActivator()
+        builder = ps_act.build_activate(td)
```

Comment:
This instance of the `builder` variable doesn't seem to be utilized, should this line be changed like what's shown below? 
```suggestion
        ps_act.build_activate(td)
```

Reply:
To be fair to myself, I just cut'n'paste this code from elsewhere at the time.
It looks like Ken has been changing things around a bit recently.
I'll revisit.

Reply:
`builder` wasn't used in the previous variant (I cut'n'paste) but it is now.  I've reworked this code to follow in the same style as the new code.